### PR TITLE
Update Docker Supervisord Initialization Spinner and Supervisord Vol Name

### PR DESCRIPTION
**Pull request by maysunfaisal**
**What type of PR is this?**
/kind bug
/kind cleanup

**What does does this PR do / why we need it**:
- Adds a spinner when pulling supervisord odo init image and initializing the supervisord volume
- Updates the supervisord volume name to have 4 random characters, to avoid name conflict on a new version of supervisord volume

**Which issue(s) this PR fixes**:

#3015 #3033

**How to test changes / Special notes to the reviewer**:
```
$ odo push

Validation
 ✓  Validating the devfile [75341ns]

Initialization
 ✓  Initializing the component [1s]   <------ new spinner

Creating Docker resources for component mynode
 ✓  Pulling image quay.io/eclipse/che-nodejs10-ubi:nightly [687ms]
 ✓  URL 127.0.0.1:54439 created
 ✓  Starting container for quay.io/eclipse/che-nodejs10-ubi:nightly [411ms]

Syncing to component mynode
 ✓  Checking files for pushing [2ms]
 ✓  Syncing files to the component [253ms]

Executing devfile commands for component mynode
 ✓  Executing devbuild command "npm install" [3s]
 ✓  Executing devrun command "nodemon app.js" [1s]

Pushing devfile component mynode
 ✓  Changes successfully pushed to component
```

```
$ docker volume ls
DRIVER              VOLUME NAME
local               odo-project-source-mynode-xmgx
local               odo-project-source-myspr1-fzjp
local               odo-supervisord-shared-data-vol-tbmt <------ Supervisord vol has 4 random chars
local               springbootpvc-myspr1-opls
```

### CODE REVIEW

File: pkg/devfile/adapters/docker/utils/utils.go [hunk 1]
```diff
@@ -234,17 +234,27 @@ func UpdateComponentWithSupervisord(comp *common.DevfileComponent, runCommand co
 // CreateAndInitSupervisordVolume creates the supervisord volume and initializes
 // it with supervisord bootstrap image - assembly files and supervisord binary
 func CreateAndInitSupervisordVolume(client lclient.Client) (string, error) {
+	log.Info("\nInitialization")
+	s := log.Spinner("Initializing the component")
+	defer s.End(false)
+
+	randomChars := util.GenerateRandomString(4)
```
Comment (johnmcollier) [inline pkg/devfile/adapters/docker/utils/utils.go:241]: Can you use `storage.GenerateVolNameFromDevfileVol` to create the volume name (name should probably be updated too 😆

File: pkg/devfile/adapters/docker/utils/utils.go [hunk 2]
```diff
@@ -234,17 +234,27 @@ func UpdateComponentWithSupervisord(comp *common.DevfileComponent, runCommand co
...
+	supervisordVolumeName, err := util.NamespaceOpenShiftObject(adaptersCommon.SupervisordVolumeName, randomChars)
+	if err != nil {
+		return "", errors.Wrapf(err, "unable to create namespaced name")
```
Comment (johnmcollier) [inline pkg/devfile/adapters/docker/utils/utils.go:244]: More specific error message

File: pkg/devfile/adapters/docker/utils/utils.go (hunk 1 above)
Comment (maysunfaisal) [reply pkg/devfile/adapters/docker/utils/utils.go]: updated

File: pkg/devfile/adapters/docker/utils/utils.go (hunk 2 above)
Comment (maysunfaisal) [reply pkg/devfile/adapters/docker/utils/utils.go]: updated

File: pkg/devfile/adapters/docker/component/utils.go [hunk 3]
```diff
@@ -24,8 +24,13 @@ import (
 	"github.com/openshift/odo/pkg/log"
 )
 
-// LocalhostIP is the IP address for localhost
-var LocalhostIP = "127.0.0.1"
+// // LocalhostIP is the IP address for localhost
+// var LocalhostIP = "127.0.0.1"
+
+const (
+	localhostIP             = "127.0.0.1"
```
Comment (johnmcollier) [inline pkg/devfile/adapters/docker/component/utils.go:31]: 👍 for un-exporting that constnat

File: pkg/devfile/adapters/docker/storage/utils_test.go [hunk 4]
```diff
@@ -331,3 +332,54 @@ func TestProcessVolumes(t *testing.T) {
 	}
 
 }
+
+func TestGenerateVolName(t *testing.T) {
```
Comment (johnmcollier) [inline pkg/devfile/adapters/docker/storage/utils_test.go:336]: 👍

**Pull request merged**